Tests/ToolBox/RandomDataGenerator/RandomDataGeneratorProgramTests.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+//using System.Linq;
+using NUnit.Framework;
+//using QuantConnect.Brokerages;
+//using QuantConnect.Securities;
+using QuantConnect.ToolBox.RandomDataGenerator;
+
+namespace QuantConnect.Tests.ToolBox.RandomDataGenerator
+{
+    [TestFixture]
+    public class RandomDataGeneratorProgramTests
+    {
+        [Test]
+        [TestCase("2020, 1, 1", "2020, 1, 1")]
+        [TestCase("2020, 1, 1", "2020, 2, 1")]
+        [TestCase("2020, 1, 1", "2020, 6, 1")]
+        [TestCase("2019, 1, 1", "2020, 6, 1")]
+        public void NextRandomGeneratedData(DateTime start, DateTime end)
+        {
+            var delistIsLessThanEnd = true;
+            var startIsLessThanEnd = true;
+            var randomValueGenerator = new RandomValueGenerator();
+            var delistDate = RandomDataGeneratorProgram.GetDelistDate(start, end, randomValueGenerator);
+
+            //start must be less than or equal to end
+            if(start > end)
+            {
+                startIsLessThanEnd = false;
+            }
```

Comment:
Not sure about this check, you're asserting the test arguments which should be okay already. Suggest directly adding expected delisting date and assert on it: `Assert.AreEqualt(expectedDelistingDate, delistingDate);`, will make the test more clear 👍 
nit - should remove unused imports `//using` & missing license header file

Reply:
I added a unit test for the midpoint value between start and end. It is difficult to directly create a TestCase for expected delist date as it is randomly generated. However, a test is done to check if the randomly generated value is between the midpoint and end.